pkgs/development/python-modules/mlflow/default.nix

```diff
@@ -0,0 +1,98 @@
+{ stdenv, buildPythonPackage, fetchPypi, writeText, lib
+, alembic
+, click
+, cloudpickle
+, requests
+, six
+, flask
+, numpy
+, pandas
+, python-dateutil
+, protobuf
+, GitPython
+, pyyaml
+, querystring_parser
+, simplejson
+, docker
+, databricks-cli
+, entrypoints
+, sqlparse
+, sqlalchemy
+, gorilla
+, gunicorn
+, pytest
+# optional, but included as override does not play nicely with toPythonApplication
+, boto3, mysqlclient 
+}:
+
+buildPythonPackage rec {
+  pname = "mlflow";
+  version = "1.4.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "9116d82be380c32fa465049d14b217c4c200ad11614f4c6674e6b524b2935206";
+  };
+  
+  patchPhase = ''
+    substituteInPlace mlflow/utils/process.py --replace \
+      "child = subprocess.Popen(cmd, env=cmd_env, cwd=cwd, universal_newlines=True," \
+      "cmd[0]='$out/bin/gunicornMlflow'; child = subprocess.Popen(cmd, env=cmd_env, cwd=cwd, universal_newlines=True,"
+  '';
+
+  gunicornScript = writeText "gunicornMlflow"
+  ''
+      #!/usr/bin/env python
+      import re
+      import sys
+      from gunicorn.app.wsgiapp import run
+      if __name__ == '__main__':
+        sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', ''', sys.argv[0])
+        sys.exit(run())
+    '';
+
+    postInstall = ''
+      gpath=$out/bin/gunicornMlflow
+      cp ${gunicornScript} $gpath
+      chmod 555 $gpath
+  '';
```

Comment:
this should be unneeded

Reply:
Even if I move out of `python-modules` and use `buildPythonApplication` this is still needed. Without, `gunicorn` will not be able to `import mlflow`, which is necessary.

Note that this is a subtle hack--I'm leveraging https://github.com/NixOS/nixpkgs/blob/731c82a447d916e78f31642ad757a3ce8911f1ea/pkgs/development/interpreters/python/wrap.sh#L52

Reply:
@FRidh thoughts?

Reply:
@FRidh the quick context is that mlflow calls `gunicorn` executable via a subprocess. `gunicorn` in turn needs to `import mlflow`. Current workaround is to make a new wrapper for `gunicorn`

Reply:
another option would be to have something like a `mflow-server` package outside of python-modules which then does gunicorn + mlflow logic

Reply:
and leave `python-modules/mlflow/default.nix` to be a vanilla python package

Reply:
now implemented